### PR TITLE
Add a link back to GitHub repo for SPECs

### LINKED ISSUE #161: Link back to repo from SPEC website
From https://scientific-python.org/specs/ , it is not easier to come back to this GitHub repo. I propose to add a link so people can come to this repo more easily from that site.

### PR DESCRIPTION
**Pull request by pllim**
Fix #161

### CODE REVIEW

File: _index.md [hunk 1]
```diff
@@ -13,3 +13,4 @@ Community discussions take place on the
 [`SPECs` Discourse forum](https://discuss.scientific-python.org/c/specs/6).
 
 Contributors must adhere to our [code of conduct]({{< relref "/code_of_conduct.md" >}}).
+To edit a SPEC, please propose a pull request to the [SPECs GitHub repository](https://github.com/scientific-python/specs).
```
Comment (stefanv) [inline _index.md:16]:
```suggestion

SPEC development takes place at https://github.com/scientific-python/specs.
```

**Comment by stefanv** [comment]
I suggested a wording tweak, but happy to merge as-is too.

**Comment by pllim** [comment]
Applied your suggestion. Thanks!

**Review by stefanv — APPROVED** [review]

**Review by jarrodmillman — APPROVED** [review]

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**